testing/trino-tests/src/test/java/io/trino/tests/AbstractTestEngineOnlyQueries.java

```diff
@@ -6044,6 +6044,10 @@ public void testLargePivot()
 
     private static String pivotQuery(int columnsCount)
     {
+        String tableValues = IntStream.range(0, columnsCount)
+                .mapToObj(columnNumber -> "regionkey")
```

Comment:
nit: `Collections.nCopies` is your friend.

most importantly though, we should use different column; using same columns many times may cause the optimizer to dedup something. 

i think we should define a regular table with many many columns.

also, this should be a prep commit.

Reply:
@hashhar I have dealt with this test in one of my recent PRs (pending review currently).
The only practical difference between yours solution and mine is that I wrapped the column reference in a function call.

Here's the relevant commit from my PR in case you want to try it out:
https://github.com/trinodb/trino/pull/7631/commits/3d0bdae8bdb32fd31063dd028353039bcd015db6

@findepi I consider this change sufficient. even if the repeated fields are somehow deduplicated, the array still refers to a table column, and so it cannot be materialized by the `ExpressionInterpreter`. And moving it out of the `ExpressionInterpreter` _should_ be enough for the query to fail. 
(In fact, currently the arrays don't work as I described above, but hopefully it will soon be fixed via https://github.com/trinodb/trino/pull/7275)

Also, I am aware that this test reflects an actual use case wich caused an issue. However, with this shape ot the pivot query, it gives no information of which array constructor actually failed. We can only learn that either of them failed. As an addition to this test case, there should be two separate cases, one involving an array of literals (passing), and one involving a "correlated" array (failing).

Reply:
Thanks @kasiafi. I've cherry-picked your commit.

Regarding the test improvement I can create an issue so that we can follow up on it.